server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -1918,10 +1918,20 @@ public boolean isUpgrading() {
   }
 
   void initializeBalancer() {
-    var localTabletBalancer = Property.createInstanceFromPropertyName(getConfiguration(),
-        Property.MANAGER_TABLET_BALANCER, TabletBalancer.class, new TableLoadBalancer());
-    localTabletBalancer.init(balancerEnvironment);
-    tabletBalancer = localTabletBalancer;
+    try {
+      getContext().getPropStore().getCache().removeAll();
+      getConfiguration().invalidateCache();
+      log.debug("Attempting to reinitialize balancer using class {}", getConfiguration().get(Property.MANAGER_TABLET_BALANCER));
+      var localTabletBalancer = Property.createInstanceFromPropertyName(getConfiguration(),
+              Property.MANAGER_TABLET_BALANCER, TabletBalancer.class, new DoNothingBalancer());
+      localTabletBalancer.init(balancerEnvironment);
```

Comment:
When running the new IT found that if init() threw an exception that it would not fall back to the DoNothingBalancer, so refactored this code to get the fallback working.